src/Compilers/Core/VBCSCompilerTests/ExecutingTestUtils.cs

```diff
@@ -0,0 +1,117 @@
+                {
+                    var key = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\MSBuild\ToolsVersions\14.0", false);
+
+                    if (key != null)
+                    {
+                        var toolsPath = key.GetValue("MSBuildToolsPath");
+                        if (toolsPath != null)
+                        {
+                            s_msbuildDirectory = toolsPath.ToString();
+                        }
+                    }
+                }
+                return s_msbuildDirectory;
+            }
+        }
+
+        public static string MSBuildExecutable { get; } = Path.Combine(MSBuildDirectory, "MSBuild.exe");
+
+        public static readonly string s_workingDirectory = Directory.GetCurrentDirectory();
+        public static string ResolveAssemblyPath(string exeName)
+        {
+            var path = Path.Combine(s_workingDirectory, exeName);
+            if (File.Exists(path))
+            {
+                return path;
+            }
+            else
+            {
+                path = Path.Combine(MSBuildDirectory, exeName);
+                return File.Exists(path) ? path : null;
+            }
+        }
+
+        public static readonly string s_compilerServerExecutableSrc = ResolveAssemblyPath(CompilerServerExeName);
+        public static readonly string s_buildTaskDllSrc = ResolveAssemblyPath(BuildTaskDllName);
+        public static readonly string s_csharpCompilerExecutableSrc = ResolveAssemblyPath("csc.exe");
+        public static readonly string s_basicCompilerExecutableSrc = ResolveAssemblyPath("vbc.exe");
+        public static readonly string s_microsoftCodeAnalysisDllSrc = ResolveAssemblyPath("Microsoft.CodeAnalysis.dll");
+        public static readonly string s_systemCollectionsImmutableDllSrc = ResolveAssemblyPath("System.Collections.Immutable.dll");
+
+        // The native client executables can't be loaded via Assembly.Load, so we just use the
+        // compiler server resolved path
+        public static readonly string s_clientExecutableBasePath = Path.GetDirectoryName(s_compilerServerExecutableSrc);
+
+
+        public static readonly string[] s_allCompilerFiles =
+        {
+            s_csharpCompilerExecutableSrc,
+            s_basicCompilerExecutableSrc,
+            s_compilerServerExecutableSrc,
+            s_microsoftCodeAnalysisDllSrc,
+            s_systemCollectionsImmutableDllSrc,
+            s_buildTaskDllSrc,
+            ResolveAssemblyPath("System.Reflection.Metadata.dll"),
+            ResolveAssemblyPath("Microsoft.CodeAnalysis.CSharp.dll"),
+            ResolveAssemblyPath("Microsoft.CodeAnalysis.VisualBasic.dll"),
+            Path.Combine(s_clientExecutableBasePath, CompilerServerExeName + ".config"),
+            Path.Combine(s_clientExecutableBasePath, "csc.rsp"),
+            Path.Combine(s_clientExecutableBasePath, "vbc.rsp")
+        };
+
+        public class TempExecutionEnvironment {
```

Comment:
`TempDirectory` doesn't implement `Disposable`, but it has a private field `TempRoot` that _does_.  

I don't think there's a way to directly dispose the tempDirectory.